pkg/awsutils/awsutils.go

```diff
@@ -1325,9 +1379,28 @@ func (cache *EC2InstanceMetadataCache) getFilteredListOfNetworkInterfaces() ([]*
 	networkInterfaces := make([]*ec2.NetworkInterface, 0)
 	for _, networkInterface := range result.NetworkInterfaces {
 		// Verify the description starts with "aws-K8S-"
-		if strings.HasPrefix(aws.StringValue(networkInterface.Description), eniDescriptionPrefix) {
-			networkInterfaces = append(networkInterfaces, networkInterface)
+		if !strings.HasPrefix(aws.StringValue(networkInterface.Description), eniDescriptionPrefix) {
+			continue
+		}
+		// Check that it's not a newly created ENI
+		tags := getTags(networkInterface, aws.StringValue(networkInterface.NetworkInterfaceId))
+
+		if value, ok := tags[eniCreatedAtTagKey]; ok {
+			parsedTime, err := time.Parse(time.RFC3339, value)
+			if err != nil {
+
+			}
+			if time.Since(parsedTime) < (5 * time.Minute) {
+				log.Infof("Found ENI created less than 5 mins so not cleaning up")
+				continue
+			}
+			log.Debugf("%v", value)
+		} else {
+			//Set time if we didn't have one. This is to catch the v1.5.x or earlier CNI versions.
```

Comment:
Nit: 
```
// Set a time if we didn't find one. This is to prevent accidentally deleting ENIs that are in the 
// process of being attached by CNI versions v1.5.x or earlier.
````